asf_public_discourse_home_decarbonisation/pipeline/faqs_identification/extract_questions.py

```diff
@@ -0,0 +1,445 @@
+    """
+    Extracts and returns a list of sentences with specific "I don't know" phrases.
+
+    The function searches for sentences that contain any of a predefined set of phrases indicating uncertainty or lack of knowledge (e.g., "don’t know", "do not know", "do not know how to"). It then puts these into a list.
+
+    This can be particularly useful for analysing text where it's important to both identify expressions of uncertainty and examine the context without these expressions.
+
+    Args:
+        sentences (List[str]): A list of sentences to be evaluated. Each item in the list should be a string representing a single sentence.
+        inclusion_phrases (List[str]): A list of inclusion phrases. Each item in the list should be a string representing a phrase to search for in the sentences.
+
+    Returns:
+        idk_phrases (List[str]):
+            - This list contains sentences that include any of the predefined "don't know" phrases.
+
+    Note:
+        - The function is case-insensitive when searching for the predefined phrases.
+    """
+    idk_phrases = []
+    for sentence in sentences:
+        if any(phrase in sentence.lower() for phrase in inclusion_phrases):
+            idk_phrases.append(sentence)
+    return idk_phrases
+
+
+def extract_X_from_idk(
+    idk_phrases: List[str], inclusion_phrases: List[str]
+) -> List[str]:
+    """
+    Extracts and returns a list of sentences without the inclusion phrases found in the "I don't know" phrases.
+
+    The function takes a list of "I don't know" phrases and a list of inclusion phrases. It searches for sentences in the "I don't know" phrases that contain any of the inclusion phrases. It then extracts the text following the inclusion phrase and returns a list of these extracted sentences.
+
+    Args:
+        idk_phrases (List[str]): A list of "I don't know" phrases. Each item in the list should be a string representing a single sentence.
+        inclusion_phrases (List[str]): A list of inclusion phrases. Each item in the list should be a string representing a phrase to search for in the "I don't know" phrases.
+
+    Returns:
+        sentences_without_inclusion (List[str]):
+            - This list contains sentences extracted from the "I don't know" phrases, excluding the inclusion phrases.
+
+    Note:
+        - The function is case-insensitive when searching for the inclusion phrases.
+        - If an inclusion phrase is not found in a sentence, that sentence will not be included in the output list.
+    """
+    sentences_without_inclusion = []
+    for sentence in idk_phrases:
+        for phrase in inclusion_phrases:
+            if phrase.lower() in sentence.lower():
+                start_index = sentence.lower().index(phrase.lower()) + len(phrase)
+                extracted_text = sentence[start_index:].strip()
+                sentences_without_inclusion.append(extracted_text)
+                break
+    return sentences_without_inclusion
```

Comment:
Just making it a little cleaner and easier to understand.

```suggestion
    sentences_without_inclusion = []
   
    idk_phrases = [phrase.lower() for phrase in idk_phrases]
    inclusion_phrases = [phrase.lower() for phrase in inclusion_phrases]
    
    for sentence in idk_phrases:
        for expression in inclusion_phrases:
            if expression in sentence:
                start_index = sentence.index(expression) + len(expression)
                extracted_text = sentence[start_index:].strip()
                sentences_without_inclusion.append(extracted_text)
                break
    return sentences_without_inclusion
```